source/fundamentals/gridfs.txt

```diff
@@ -4,7 +4,371 @@ GridFS
+
+   The ``Find()`` method requires a query filter as a parameter. To match all 
+   documents in the ``files`` collection, pass an empty query filter to ``Find()``.
+
+The following example retrieves the file name and length of documents in the ``files``
+collection. It passes a filter argument to the ``Find()`` method to match documents with
+``length`` values greater than ``1500``:
+
+.. code-block:: go
+
+   filter := bson.D {{"length", bson.D{{"$gt", 1500}}}
+   cursor, err := bucket.Find(filter)
+
+   type gridfsFile struct {
+      Name   string `bson:"filename"`
+      Length int64  `bson:"length"`
+   }
+   var foundFiles []gridfsFile
+   if err = cursor.All(context.TODO(), &foundFiles); err != nil {
+      log.Fatal(err)
+   }
+
+   for _, file := range foundFiles {
+      fmt.Printf("filename: %s, length: %d\n", file.Name, file.Length)
+   }
+
+This code outputs the ``name`` and ``length`` values of documents in the ``files``
+collection that match the query filter.
+
+.. _golang-download-files:
+
+Download Files
+~~~~~~~~~~~~~~
+
+You can download a GridFS file in one of the following ways:
+
+- Use the ``DowloadToStream()`` method to download a file to an output stream
+- Use the ``OpenDownloadStream()`` method to open an input stream
+
+Download a File to an Output Stream
+```````````````````````````````````
+
+You can download a file in a GridFS bucket directly to an output stream using the
+``DownloadToStream()`` method. ``DownloadToStream()`` takes an ``ObjectID`` value and an
+``io.Writer`` as parameters. The method downloads the file with the specified ``ObjectID``
+and writes to the ``io.Writer``.
+
+The following example downloads a file with an ``_id`` value of ``62e005408bc`` and writes
+to a file buffer:
+
+.. code-block:: go
+
+   id, err = primitive.ObjectIDFromHex("62e005408bc")
+   fileBuffer := bytes.NewBuffer(nil)
+   if _, err := bucket.DownloadToStream(id, fileBuffer); err != nil {
+      log.Fatal(err)
+   }
+
+Download a File to an Input Stream
+``````````````````````````````````
+
+You can download a file in a GridFS bucket to memory with an input stream. Call
+the ``OpenDownloadStream()`` method on the GridFS bucket. ``OpenDownloadStream()`` takes
+a ``ObjectID`` value as a parameter and returns an input stream from which you can read
+the file.
+
+The following example downloads a file with an ``_id`` value of ``62e005408bc`` into memory
+and reads its contents:
+
+.. code-block:: go
+
+   id, err = primitive.ObjectIDFromHex("62e005408bc")
+   downloadStream, err := bucket.OpenDownloadStream(id)
+   if err != nil {
+      log.Fatal(err)
+   }
+
+   fileBuffer := bytes.NewBuffer(nil)
+   if _, err := downloadStream.Read(fileBuffer); err != nil {
+      log.Fatal(err)
+   }
+
+.. _golang-rename-files:
+
+Rename Files
+~~~~~~~~~~~~
+
+You can update the name of a GridFS file in your bucket by calling the ``Rename()``
+method on the GridFS bucket. Pass an ``ObjectID`` value and a new ``filename`` value as
```

Comment:
I think the wording here should stay consistent with the previous sections.

```suggestion
You can update the name of a GridFS file in your bucket by using the ``Rename()``
method. Pass an ``ObjectID`` value and a new ``filename`` value as
```

Reply:
Made this change